src/PerfView/PerfViewData.cs

```diff
@@ -5098,6 +5097,33 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
                 eventSource.Kernel.AddCallbackForEvents<ObjectDuplicateHandleTraceData>(data => onHandleEvent(data.ObjectTypeName, data.Object, data.TargetHandle, data.TargetProcessID, data));
                 eventSource.Process();
             }
+            else if (streamName.StartsWith("Processor"))
+            {
+                eventSource.AllEvents += delegate (TraceEvent data)
```

Comment:
Rather than hooking this up to `eventSource.AllEvents`, you should hook this up to `eventSource.Kernel.PerfInfoSample`.  Then, you can change the argument type on the delegate from `TraceEvent` to `SampledProfileTraceData` and you no longer have to process all events or do the cast.

Reply:
Is SampledProfileTraceData guaranteed to have a valid stack (data.CallStackIndex() != CallStackIndex.Invalid) ? If it does, then I can also remove the check for invalid stack.
I'll leave the check to be defensive for now if that's not true.

Reply:
It's good to leave it there - it is possible to be missing a stack.